server/src/db/catalog.rs

```diff
@@ -119,6 +128,37 @@ impl Catalog {
             .cloned()
             .context(UnknownPartition { partition_key })
     }
+
+    /// Returns all chunks within the catalog in an arbitrary order
+    pub fn chunks(&self) -> Vec<Arc<RwLock<Chunk>>> {
+        let mut chunks = Vec::new();
+        let partitions = self.partitions.read();
+
+        for partition in partitions.values() {
+            let partition = partition.read();
+            chunks.extend(partition.chunks().cloned())
+        }
+        chunks
```

Comment:
```suggestion
        let partitions = self.partitions.read();

        partitions.values()
          .flat_map(|partition| {
            let partition = partition.read();
            (partition.chunks().cloned()
          })
        .collect()
```

You could even avoid the `collect` if you wanted to change the signature to 

```
    pub fn chunks(&self) -> impl Iterator<Item=Arc<RwLock<Chunk>>> {
```

Reply:
I don't think you can as the iterator returned by partition.chunks().cloned() needs to outlive the closure body, but it cannot outlive the lock guard which lives as long as the closure body